### PR TITLE
Fixes #74 #67 - Classpath-scanning is now optional and faster

### LINKED ISSUE #74: FastClasspathScanner is outdated -- consider porting to ClassGraph
Your project, `mbknor/mbknor-jackson-jsonSchema`, depends on the outdated library FastClasspathScanner in the following source files:

* [`src/main/scala/com/kjetland/jackson/jsonSchema/JsonSchemaGenerator.scala`](https://github.com/mbknor/mbknor-jackson-jsonSchema/tree/master/src/main/scala/com/kjetland/jackson/jsonSchema/JsonSchemaGenerator.scala)

FastClasspathScanner has been significantly reworked since the version your code depends upon:

* a significant number of bugs have been fixed
* some nontrivial API changes have been made to simplify and unify the API
* FastClasspathScanner has been renamed to ClassGraph: https://github.com/classgraph/classgraph

ClassGraph is a significantly more robust library than FastClasspathScanner, and is more future-proof. All future development work will be focused on ClassGraph, and FastClasspathScanner will see no future development.

Please consider porting your code over to the new ClassGraph API, particularly if your project is in production or has downstream dependencies:

* https://github.com/classgraph/classgraph/wiki/Porting-FastClasspathScanner-code-to-ClassGraph
* https://github.com/classgraph/classgraph/wiki/Code-examples
* https://github.com/classgraph/classgraph/wiki/ClassGraph-API

Feel free to close this bug report if this code is no longer in use. (You were sent this bug report because your project depends upon FastClasspathScanner, and has been starred by 69 users. Apologies if this bug report is not helpful.)

### PR DESCRIPTION
**Pull request by mbknor**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -346,6 +346,27 @@ And here is the equivalent for Java:
...
+
+This is how to configure JsonSchemaGenerator to use reflection:
+
+```
+    // Scan entire classpath
+    config = config.withReflection( new ClassGraph().enableClassInfo().scan() )
+    
+    // Scan only some packages (this is faster)
+    config = config.withReflection( new ClassGraph().enableClassInfo().whitelistPackages("com.xyz").scan() )
+
```
[the remark this answers is not in the document]
Comment (mbknor) [reply README.md]: Very good idea!

I'l make a trait that is used to resove it.. And create a default impl for ScanResult. Then user can chose other impl. I'll look into it

[the remark this answers is not in the document]
Comment (zrrobbins) [reply README.md]: Yeah that's a good idea! Another option could be to just take a list of classes or a packages, and you can still use `ClassGraph`, but the user won't need to.

File: src/main/scala/com/kjetland/jackson/jsonSchema/JsonSchemaGenerator.scala [hunk 2]
```diff
@@ -687,17 +695,21 @@ class JsonSchemaGenerator
...
+              getReflection.getSubclasses(_type.getRawClass.getName).loadClasses().asScala.toList
 
             case x => throw new Exception("We do not support @jsonTypeInfo.use = " + x)
           }
 
       }.getOrElse(List())
     }
 
+    private def getReflection():ScanResult = {
+      config.reflection.getOrElse( throw new Exception(s"Reflection(io.github.classgraph.ScanResult) not configured. https://github.com/mbknor/mbknor-jackson-jsonSchema#using-reflection"))
```
[the remark this answers is not in the document]
Comment (zrrobbins) [reply src/main/scala/com/kjetland/jackson/jsonSchema/JsonSchemaGenerator.scala]: Yeah I think that's an awesome idea!

File: src/main/scala/com/kjetland/jackson/jsonSchema/JsonSchemaGenerator.scala [hunk 3]
```diff
@@ -157,8 +159,14 @@ case class JsonSchemaConfig
   useMultipleEditorSelectViaProperty:Boolean, // https://github.com/jdorn/json-editor/issues/709
   uniqueItemClasses:Set[Class[_]], // If rendering array and type is instanceOf class in this set, then we add 'uniqueItems": true' to schema - See // https://github.com/jdorn/json-editor for more info
   classTypeReMapping:Map[Class[_], Class[_]], // Can be used to prevent rendering using polymorphism for specific classes.
-  jsonSuppliers:Map[String, Supplier[JsonNode]] // Suppliers in this map can be accessed using @JsonSchemaInject(jsonSupplierViaLookup = "lookupKey")
-)
+  jsonSuppliers:Map[String, Supplier[JsonNode]], // Suppliers in this map can be accessed using @JsonSchemaInject(jsonSupplierViaLookup = "lookupKey")
+  reflection:Option[ScanResult] = None
```
Comment (zrrobbins) [inline src/main/scala/com/kjetland/jackson/jsonSchema/JsonSchemaGenerator.scala]: This also should be added to the Java API so Java users like me can access it :)

**Pull request merged**